src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -551,13 +551,35 @@ void MPS::apply_swap_internal(uint_t index_A, uint_t index_B, bool swap_gate) {
     // and the qubit at index_B (or index_A+1) is moved one position 
     //to the left
     std::swap(qubit_ordering_.order_[index_A], qubit_ordering_.order_[index_B]);    
+    // For logging purposes:
+    if (MPS::get_mps_log_data()) {
```

Comment:
How about checking the log level only inside the function, like you do in the previous diff

Reply:
Did as you suggested.